### ISSUE TITLE
[custom devices] Add release / profile mode support

### ISSUE THREAD

**Issue by ardera** [flutter/flutter#79681]
Currently custom devices only support debug mode. In the future, we could look at enabling release & profile mode.

This depends on #75823 (linux ARMv7 support).

Running the app is easily implemented:
- add `runRelease` and `runProfile` command in custom devices config (behaviour similiar to `runDebug` command)
- document those in the JSON schema

For building, we need ARMv7 artifacts (`gen_snapshot`, ideally also flutter engines). I'm not sure what layout other embedders expect, but for flutter-pi the finished `app.so` should be contained in the asset bundle.

**Comment by jonahwilliams** [comment]
We'd potentially need to support targeting either 32 or 64 bit arm - though really there is nothing stopping a custom device from being windows/linux.

gen_snapshot is compiled for a specified host and target platform, but we don't make it easy to download an arbitrary combination since they are bundled up with each platform's artifacts. Something to consider is what sorts of cross compilation scenarios we need to support, and whether there is a way we could start small with what we have today and gradually add support for more platforms.

**Comment by ardera** [comment]
> We'd potentially need to support targeting either 32 or 64 bit arm - though really there is nothing stopping a custom device from being windows/linux.

agree, specifying the target arch itself should be resolved by #78151.

> gen_snapshot is compiled for a specified host and target platform, but we don't make it easy to download an arbitrary combination since they are bundled up with each platform's artifacts. Something to consider is what sorts of cross compilation scenarios we need to support, and whether there is a way we could start small with what we have today and gradually add support for more platforms.

Sure, I think what makes this easier is that we don't have any native C/C++ plugin code we need to compile. So we don't need a cross compiler or a sysroot, only the correct `gen_snapshot` (from the artifacts bundle) and it should just work.
